site/content/integrate/slash-commands/custom/_index.md

```diff
@@ -0,0 +1,137 @@
+
+      {{<note "Note:">}}
+   [Enable integrations to override usernames](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-usernames) must be set to `true` in `config.json` to override usernames. Enable them from **System Console > Integrations > Integration Management**, or ask your System Admin to do so. If not enabled, the username is set to `webhook`.
+
+   Similarly, [Enable integrations to override profile picture icons](https://docs.mattermost.com/configure/configuration-settings.html#enable-integrations-to-override-profile-picture-icons) must be set to `true` in `config.json` to override profile picture icons. Enable them from **System Console > Integrations > Integration Management**, or ask your System Admin to do so. If not enabled, the icon of the creator of the webhook URL is used to post messages.
+      {{</note>}}
+
+3. Select **Save**. On the next page, copy the **Token** value. This will be used in a later step.
+
+   ![image](slash_commands_token.png)
+
+4. Next, write your external application. Include a function which receives HTTP `POST` or HTTP `GET` requests from Mattermost. The request will look something like this:
+
+    ```http request
+    POST /slash-command HTTP/1.1
+    Host: example.com
+    Accept: application/json
+    Accept-Encoding: gzip
+    Authorization: Token qzgakf1nx3yt9dr4n8585ihbxy
+    Content-Length: 313
+    Content-Type: application/x-www-form-urlencoded
+    User-Agent: Mattermost-Bot/1.1
+    
+    channel_id=jux16pkewjrkfj3ehep1psxyxc&
+    channel_mentions=["saepe-5", "aut-8"]&
+    channel_mentions_ids=["r3j6sby343fpfdxcbwqg95rfsa", "ehjj46yk7ifptr5bpfb966s6mc"]&
+    channel_name=town-square&
+    command=%2Ftest&
+    response_url=http%3A%2F%2Flocalhost%3A8065%2Fhooks%2Fcommands%2Fxbrkb8p393gjpq5cawei7npije&
+    team_domain=test&
+    team_id=carya1qs77bemjup96ff538snh&
+    text=asd&
+    token=okwexkjpe7ygb8eq1ww58t483w&
+    user_id=aoa1agao6t8fmx3ikt1j9w5ybw&
+    user_mentions=["aaron.peterson", "aaron.medina"]&
+    user_mentions_ids=["q5s3b7xzgprp5eid8h66j9epsy", "czwmumrmw7dfxecww7qibkkoor"]
+    user_name=somename&
+    ```
+
+   If your integration sends back a JSON response, make sure it returns the `application/json` content-type.
+
+5. Add a configurable *MATTERMOST_TOKEN* variable to your application and set it to the **Token** value from step 7. This value will be used by your application to confirm the HTTP `POST` or `GET` request came from Mattermost.
+6. To have your application post a message back to `town-square`, it can respond to the HTTP `POST` request with a JSON response such as:
```

Comment:
I think you should have a mention/in-page link down to the Parameters section here. You should mention this immediately now that it's relevant, it's arguably the most critical element of the page and right now it's stuffed at the bottom.